Crash in settings if SD card was removed

##### Detailed steps causing the problem:
- Have GeocacheData moved to external SD card
- Close and kill c:geo
- Remove SD card from device
- Start c:geo
- Log shows Warning (see first sdb log attached)
- Try to enter Menu -> Settings (e.g. if you plan to set the storage to internal again)

##### Actual behavior after performing these steps:
Crash

##### Expected behavior after performing these steps:
Detect absence of GeocacheData dir and e.g. fallback (to be discussed) to internal dir in that case.
Also without a fallback c:geo should not crash when entering settings menu allowing the user to manually adjust the dir setting.

##### Version of c:geo used:
2017.03.25-NB

##### Is the problem reproducible for you?
Yes

##### System information:
Android 5.1.1
Samsung Galaxy Note 10.1 Edition 2014 (SD removeable without power down)

##### Other comments and remarks:

Log after c:geo startup:
```
12:08:54.776 Warning cgeo 26754  [RxCachedThreadScheduler-2] DataStore.clean
12:08:54.776 Warning cgeo 26754  java.lang.NullPointerException: Attempt to invoke virtual method 'java.io.File java.io.File.getCanonicalFile()' on a null object reference
12:08:54.776 Warning cgeo 26754  	at android.os.Environment.getStorageVolume(Environment.java:900)
12:08:54.776 Warning cgeo 26754  	at android.os.Environment.getExternalStorageState(Environment.java:758)
12:08:54.776 Warning cgeo 26754  	at android.os.Environment.getStorageState(Environment.java:744)
12:08:54.776 Warning cgeo 26754  	at android.support.v4.os.EnvironmentCompatKitKat.getStorageState(EnvironmentCompatKitKat.java:25)
12:08:54.776 Warning cgeo 26754  	at android.support.v4.os.EnvironmentCompat.getStorageState(EnvironmentCompat.java:58)
12:08:54.776 Warning cgeo 26754  	at cgeo.geocaching.storage.LocalStorage.getAvailableExternalPrivateCgeoDirectories(LocalStorage.java:82)
12:08:54.776 Warning cgeo 26754  	at cgeo.geocaching.storage.LocalStorage.getExternalPrivateCgeoDirectory(LocalStorage.java:98)
12:08:54.776 Warning cgeo 26754  	at cgeo.geocaching.storage.LocalStorage.getGeocacheDataDirectory(LocalStorage.java:281)
12:08:54.776 Warning cgeo 26754  	at cgeo.geocaching.storage.LocalStorage.getGeocacheDataDirectory(LocalStorage.java:156)
12:08:54.776 Warning cgeo 26754  	at cgeo.geocaching.storage.DataStore$10.run(DataStore.java:2481)
12:08:54.776 Warning cgeo 26754  	at io.reactivex.Scheduler$1.run(Scheduler.java:138)
12:08:54.776 Warning cgeo 26754  	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:59)
12:08:54.776 Warning cgeo 26754  	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:51)
12:08:54.776 Warning cgeo 26754  	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
12:08:54.776 Warning cgeo 26754  	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
12:08:54.776 Warning cgeo 26754  	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
12:08:54.776 Warning cgeo 26754  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
12:08:54.776 Warning cgeo 26754  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
12:08:54.776 Warning cgeo 26754  	at java.lang.Thread.run(Thread.java:818)
```

Log when trying to enter settings:
```
12:09:41.659 Error AndroidRuntime 26950  FATAL EXCEPTION: main
12:09:41.659 Error AndroidRuntime 26950  Process: cgeo.geocaching, PID: 26950
12:09:41.659 Error AndroidRuntime 26950  java.lang.RuntimeException: Unable to start activity ComponentInfo{cgeo.geocaching/cgeo.geocaching.settings.SettingsActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.io.File java.io.File.getCanonicalFile()' on a null object reference
12:09:41.659 Error AndroidRuntime 26950  	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3124)
12:09:41.659 Error AndroidRuntime 26950  	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3233)
12:09:41.659 Error AndroidRuntime 26950  	at android.app.ActivityThread.access$1000(ActivityThread.java:197)
12:09:41.659 Error AndroidRuntime 26950  	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656)
12:09:41.659 Error AndroidRuntime 26950  	at android.os.Handler.dispatchMessage(Handler.java:102)
12:09:41.659 Error AndroidRuntime 26950  	at android.os.Looper.loop(Looper.java:145)
12:09:41.659 Error AndroidRuntime 26950  	at android.app.ActivityThread.main(ActivityThread.java:6873)
12:09:41.659 Error AndroidRuntime 26950  	at java.lang.reflect.Method.invoke(Native Method)
12:09:41.659 Error AndroidRuntime 26950  	at java.lang.reflect.Method.invoke(Method.java:372)
12:09:41.659 Error AndroidRuntime 26950  	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
12:09:41.659 Error AndroidRuntime 26950  	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
12:09:41.659 Error AndroidRuntime 26950  Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.io.File java.io.File.getCanonicalFile()' on a null object reference
12:09:41.659 Error AndroidRuntime 26950  	at android.os.Environment.getStorageVolume(Environment.java:900)
12:09:41.659 Error AndroidRuntime 26950  	at android.os.Environment.getExternalStorageState(Environment.java:758)
12:09:41.659 Error AndroidRuntime 26950  	at android.os.Environment.getStorageState(Environment.java:744)
12:09:41.659 Error AndroidRuntime 26950  	at android.support.v4.os.EnvironmentCompatKitKat.getStorageState(EnvironmentCompatKitKat.java:25)
12:09:41.659 Error AndroidRuntime 26950  	at android.support.v4.os.EnvironmentCompat.getStorageState(EnvironmentCompat.java:58)
12:09:41.659 Error AndroidRuntime 26950  	at cgeo.geocaching.storage.LocalStorage.getAvailableExternalPrivateCgeoDirectories(LocalStorage.java:82)
12:09:41.659 Error AndroidRuntime 26950  	at cgeo.geocaching.settings.SettingsActivity.initExtCgeoDirPreference(SettingsActivity.java:265)
12:09:41.659 Error AndroidRuntime 26950  	at cgeo.geocaching.settings.SettingsActivity.initPreferences(SettingsActivity.java:144)
12:09:41.659 Error AndroidRuntime 26950  	at cgeo.geocaching.settings.SettingsActivity.onCreate(SettingsActivity.java:111)
12:09:41.659 Error AndroidRuntime 26950  	at android.app.Activity.performCreate(Activity.java:6550)
12:09:41.659 Error AndroidRuntime 26950  	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1120)
12:09:41.659 Error AndroidRuntime 26950  	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3077)
12:09:41.659 Error AndroidRuntime 26950  	... 10 more
12:09:41.659 Warning ActivityManager 2919    Force finishing activity 1 cgeo.geocaching/.settings.SettingsActivity
```

--- a/main/src/cgeo/geocaching/storage/LocalStorage.java
+++ b/main/src/cgeo/geocaching/storage/LocalStorage.java
@@ -79,7 +79,7 @@ public final class LocalStorage {
         final List<File> extDirs = new ArrayList<>();
         final File[] externalFilesDirs = ContextCompat.getExternalFilesDirs(CgeoApplication.getInstance(), null);
         for (final File dir : externalFilesDirs) {
-            if (EnvironmentCompat.getStorageState(dir).equals(Environment.MEDIA_MOUNTED)) {
+            if (dir != null && EnvironmentCompat.getStorageState(dir).equals(Environment.MEDIA_MOUNTED)) {
                 extDirs.add(dir);
             }
         }

